pkg/interface/src/apps/chat/components/lib/message.js

```diff
@@ -4,6 +4,43 @@ import { uxToHex, cite, writeText } from '../../../../lib/util';
 import moment from 'moment';
 import ReactMarkdown from 'react-markdown';
 import RemarkDisableTokenizers from 'remark-disable-tokenizers';
+import SyntaxHighlighter from 'react-syntax-highlighter';
+import { darcula, atomOneLight } from 'react-syntax-highlighter/dist/esm/styles/hljs';
+
+export class CodeBlock extends Component {
+  static defaultProps = {
+    language: 'markdown',
+  }
+
+  constructor() {
+    super();
+    const scheme = window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+    this.state = {
+      scheme
+    };
+  }
+
+  componentDidMount() {
+    const darkModeMediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
```

Comment:
A media query watcher will be set up for each message being rendered. It'd be ideal to only do this once. When the groups refactor is merged, this probably should be a part of the state and handed down to the component in a prop.

Reply:
Ah, that is pretty expensive. Maybe for now, put in `ChatScreen` and pass it down. Or, just set the mode once. But then it doesn't change on MQ change, which can look weird but only for a little bit.

Reply:
yeah for now put it in ChatScreen

Reply:
Why not rebase on top of `lf/groups-refactor`, add the media plumbing as a prop and target `release/next-userspace` here? It seems odd to have a change immediately obsolesced within days ...

Reply:
Want me to implement the MQ in groups-refactor and add this component in there? Code highlighting can wait a couple days